src/Lib.P2P.Tests/PeerTest.cs

```diff
@@ -0,0 +1,163 @@
+
+        private static string marsAddress =
+            "/ip4/10.1.10.10/tcp/29087/ipfs/QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3";
+
+        [TestMethod]
+        public new void ToString()
+        {
+            Assert.AreEqual("", new Peer().ToString());
+            Assert.AreEqual(marsId, new Peer {Id = marsId}.ToString());
+        }
+
+        [TestMethod]
+        public void DefaultValues()
+        {
+            var peer = new Peer();
+            Assert.AreEqual(null, peer.Id);
+            Assert.AreEqual(0, peer.Addresses.Count());
+            Assert.AreEqual("unknown/0.0", peer.ProtocolVersion);
+            Assert.AreEqual("unknown/0.0", peer.AgentVersion);
+            Assert.AreEqual(null, peer.PublicKey);
+            Assert.AreEqual(false, peer.IsValid()); // missing peer ID
+            Assert.AreEqual(null, peer.ConnectedAddress);
+            Assert.IsFalse(peer.Latency.HasValue);
+        }
+
+        [TestMethod]
+        public void ConnectedPeer()
+        {
+            var peer = new Peer
+            {
+                ConnectedAddress = new MultiAddress(marsAddress),
+                Latency = TimeSpan.FromHours(3.03 * 2)
+            };
+            Assert.AreEqual(marsAddress, peer.ConnectedAddress.ToString());
+            Assert.AreEqual(3.03 * 2, peer.Latency.Value.TotalHours);
+        }
+
+        [TestMethod]
+        public void Validation_No_Id()
+        {
+            var peer = new Peer();
+            Assert.AreEqual(false, peer.IsValid());
+        }
+
+        [TestMethod]
+        public void Validation_With_Id()
+        {
+            Peer peer = marsId;
+            Assert.AreEqual(true, peer.IsValid());
+        }
+
+        [TestMethod]
+        public void Validation_With_Id_Pubkey()
+        {
+            var peer = new Peer
+            {
+                Id = marsId,
+                PublicKey = marsPublicKey
+            };
+            Assert.AreEqual(true, peer.IsValid());
+        }
+
+        [TestMethod]
+        public void Validation_With_Id_Invalid_Pubkey()
+        {
+            var peer = new Peer
+            {
+                Id = plutoId,
+                PublicKey = marsPublicKey
+            };
+            Assert.AreEqual(false, peer.IsValid());
+        }
+
+        [TestMethod]
+        public void Value_Equality()
+        {
+            var a0 = new Peer {Id = marsId};
+            var a1 = new Peer {Id = marsId};
+            var b = new Peer {Id = plutoId};
+            Peer c = null;
+            Peer d = null;
+
+            Assert.IsTrue(c == d);
+            Assert.IsFalse(c == b);
+            Assert.IsFalse(b == c);
+
+            Assert.IsFalse(c != d);
+            Assert.IsTrue(c != b);
+            Assert.IsTrue(b != c);
+
+#pragma warning disable 1718
+            Assert.IsTrue(a0 == a0);
+            Assert.IsTrue(a0 == a1);
+            Assert.IsFalse(a0 == b);
+
+#pragma warning disable 1718
+            Assert.IsFalse(a0 != a0);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Correct one of the identical expressions on both sides of operator '!='.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)